### PR TITLE
3128 feat/dataset feedback number of meters

### PR DESCRIPTION
**Pull request by perryr16**
#### Any background context you want to provide?
If a property file with meter data is imported, SEED generates two different import records. One for the property import, and one for the meter data import. The "# of records" listed in the view indicates the number of properties that were in the imported file. This works fine for the property import, but the meter data import returns 0 for "# of records". 

ImportFiles are SEED models representing a property file that has been imported. SEED reuses this model for meter imports for display purposes but most of the model fields are left blank. The model column `num_rows` is used to populate "# of records". This field was left as None during the meter import.

#### What's this PR do?
Attaches the rows of the proposed meter import file to the `ImportFile.num_rows` property, and updates any conditionals to prevent unexpected functionality.

#### How should this be manually tested?
- Import a Property file with meter data. Following the property import click "Import Meters from the same file". Note this may take several seconds to appear, and several seconds to prompt the next modal screen.
Screen Shot 2022-03-08 at 12 34 33 PM
- Once the proposed meter screen appears go through the meter import steps until complete
- Go to Data, click the relevant data set and there should be 2 data files generated from the recent upload. One for properties, one for meters. 
- The meter import data file should have "Data Mapping" and "Data Pairing" functionality disabled.

Test File: [ESPM_JCC_Custom_Download_BETTERTest.xlsx](https://github.com/SEED-platform/seed/files/8209288/ESPM_JCC_Custom_Download_BETTERTest.xlsx)

#### What are the relevant tickets?
#3158

#### Screenshots (if appropriate)
Screen Shot 2022-03-08 at 2 55 05 PM

Screen Shot 2022-03-08 at 1 17 59 PM

### CODE REVIEW

File: seed/static/seed/partials/dataset_detail.html [hunk 1]
```diff
@@ -41,7 +41,7 @@ <h1>{$:: dataset.name $}</h1>
                                 <td>{$:: f.created | date: 'MM/dd/yyyy hh:mm:ss a' $}</td>
                                 <td class="is_aligned_right">{$:: f.num_rows || 0 $}</td>
                                 <td>{$:: getCycleName(f.cycle) $}</td>
-                                <td><a id="data-mapping-{$:: $index $}" role="button" class="btn btn-sm btn-primary" ui-sref="mapping(::{importfile_id: f.id})" ng-disabled="::!f.num_rows">{$:: f.number_of_mappings $} {$:: 'Data Mapping' | translate $}</a></td>
+                                <td><a id="data-mapping-{$:: $index $}" role="button" class="btn btn-sm btn-primary" ui-sref="mapping(::{importfile_id: f.id})" ng-disabled="::!f.num_rows || !f.cached_second_to_fifth_row">{$:: f.number_of_mappings $} {$:: 'Data Mapping' | translate $}</a></td>
```
Comment (perryr16) [inline seed/static/seed/partials/dataset_detail.html:44]: `cache_second_to_fifth_row` is the property that is used in the following function, but will remain null for meter data ImportRecords

File: seed/static/seed/js/controllers/data_upload_modal_controller.js [hunk 2]
```diff
@@ -388,6 +390,7 @@ angular.module('BE.seed.controller.data_upload_modal', [])
         cycle_id = cycle_id || $scope.selectedCycle.id;
         $scope.uploader.in_progress = true;
         save_raw_assessed_data(file_id, cycle_id, true);
+        $rootScope.$emit('datasets_updated')
```
Comment (perryr16) [inline seed/static/seed/js/controllers/data_upload_modal_controller.js:393]: trigger `refresh_data()` in the /dataset_list_controller.js after user confirms meters

Comment (Ryoken) [reply seed/static/seed/js/controllers/data_upload_modal_controller.js:393]: nice!  could we add a semicolon?

**Comment by Ryoken** [comment]
looks good!

**Pull request merged**